File: LeetCode75_LevelTwo/Where_Will_the_Ball_Fall/Where_Will_the_Ball_Fall.cc
// 1706. Where Will the Ball Fall C++

#include <vector>
using namespace std;

class Solution {
public:
    vector<int> findBall(vector<vector<int>>& grid) {
        int rows = grid.size();
        int cols = grid[0].size();
        vector<int> output;
        for (int i = 0; i < cols; i++) {
            int row = 0;
            int col = i;
            while (true) {
                if (row == rows) {
                    output.emplace_back(col);
                    break;
                } else if (grid[row][col] == 1 && col == cols-1) {
                    output.emplace_back(-1);
                    break;
                } else if (grid[row][col] == 1 && grid[row][col + 1] == -1) {
                    output.emplace_back(-1);
                    break;
                } else if (grid[row][col] == 1) {
                    row++;
                    col++;
                } else if (grid[row][col] == -1 && col == 0) {
                    output.emplace_back(-1);
                    break;
                } else if (grid[row][col] == -1 && grid[row][col - 1] == 1) {
                    output.emplace_back(-1);
                    break;
                } else if (grid[row][col] == -1){
                    row++;
                    col--;
                }
            }
        }
        return output;
    }
};

File: LeetCode75_LevelTwo/Where_Will_the_Ball_Fall/Where_Will_the_Ball_Fall.py
# 1706. Where Will the Ball Fall Python

class Solution(object):
    def findBall(self, grid):
        """
        :type grid: List[List[int]]
        :rtype: List[int]
        """
        rows = len(grid)
        cols = len(grid[0])
        output = []
        for i in range(0, cols):
            row = 0
            col = i
            while True:
                if row == rows:
                    output.append(col)
                    break
                if grid[row][col] is 1 and col is cols - 1:
                    output.append(-1)
                    break
                elif grid[row][col] is 1 and grid[row][col + 1] is -1:
                    output.append(-1)
                    break
                elif grid[row][col] is 1:
                    row += 1
                    col += 1
                    continue
                elif grid[row][col] is -1 and col is 0:
                    output.append(-1)
                    break
                elif grid[row][col] is -1 and grid[row][col - 1] is 1:
                    output.append(-1)
                    break
                elif grid[row][col] is -1:
                    row += 1
                    col -= 1
                    continue
        return output
